src/SystemLoader.cc

```diff
@@ -85,7 +99,10 @@ class ignition::gazebo::SystemLoaderPrivate
       return false;
     }
 
-    _gzPlugin = this->loader.Instantiate(_sdfPlugin.Name());
+    std::string pluginToInstantiate = _sdfPlugin.Name().empty() ?
+        pluginName : _sdfPlugin.Name();
```

Comment:
add comment and updated function doxygen comment. I didn't add it to the tutorial since SDF still considers `name` to be a required attribute so the user will still need to specify it. The `name` arg can only be empty if loading the plugin dynamically through gazebo using the SystemLoader.